src/fifelse.c

```diff
@@ -343,3 +343,194 @@ SEXP fcaseR(SEXP na, SEXP rho, SEXP args) {
   UNPROTECT(nprotect);
   return ans;
 }
+
+SEXP topnR(SEXP vec, SEXP n, SEXP dec) {
+	int nprotect = 0;
+	int64_t i, j, idx = 0;
+	int len0 = asInteger(n);
+	const int64_t len1 = xlength(vec);
+	
+	if (isS4(vec) && !INHERITS(vec, char_nanotime)) {
+		error("S4 class objects (excluding nanotime) are not supported.");
+	}
+	if (len0 > 1000) {
```

Comment:
If input is 1e8 values it should still be faster to do `topn(x, 1500)`, no?

A fixed value I don't think makes sense. Maybe something like `len1/2` makes more sense.

Reply:
Or perhaps just re-route with warning -- do `order(x)[1:n]` if `n>length(x)/2`

Reply:
Please see my comment above. Thanks.